### PR TITLE
TBM-6040 - added protection from deep data properties on interaction editor

### PR DESCRIPTION
**Pull request by troyblank**

### CODE REVIEW

File: components/interactionEditor/contextBar/contextBar.js [hunk 1]
```diff
@@ -63,6 +64,19 @@ export default class ContextBar extends Component {
 
         const flagColor = flagIsActive ? 'orange' : 'black';
         const campaignNames = exports.extractCampaignNames(contextBarData.campaignIds);
+        const observer = objectPath.get(contextBarData, 'observation.observer.analyst');
```
Comment (jhweaver) [inline components/interactionEditor/contextBar/contextBar.js]: This is actually the agent. I'm going to update interactions so you can get the name you're looking for here from observation.observer

Comment (troyblank) [inline components/interactionEditor/contextBar/contextBar.js]: Alright, Mr @jhweaver this change has been made to the data structure. Thanks for this insight.

File: components/interactionEditor/contextBar/contextBar.js [hunk 2]
```diff
@@ -63,6 +64,19 @@ export default class ContextBar extends Component {
 
         const flagColor = flagIsActive ? 'orange' : 'black';
         const campaignNames = exports.extractCampaignNames(contextBarData.campaignIds);
+        const observer = objectPath.get(contextBarData, 'observation.observerName');
+        const totalHoldTime = objectPath.get(contextBarData, 'observation.totalHoldTime');
+        const lastSentiment = objectPath.get(contextBarData, 'observation.lastSentiment');
+        let observationDate = objectPath.get(contextBarData, 'observation.observer.observingDate');
```
Comment (jhweaver) [inline components/interactionEditor/contextBar/contextBar.js]: We ended up removing the redundant "observation.observer" The date is now just "observation.observationDate" (note the "observING" -> "observaTION")

File: assets/js/mockData/interactionObservedData.js [hunk 3]
```diff
@@ -6,8 +6,8 @@ const interactionData = {
     observation: {
         isObserved: true,
         lastSentiment: 'Indifferent',
+        observerName: 'Theresa Edwards',
         observer: {
-            analyst: 'Theresa Edwards',
             observingDate: '2017-01-01T19:31:19.741000',
```
Comment (jhweaver) [inline assets/js/mockData/interactionObservedData.js]: We moved this out of observation.observer to observation, and renamed it to observationDate

File: assets/js/mockData/interactionObservedData.js [hunk 4]
```diff
@@ -6,8 +6,8 @@ const interactionData = {
     observation: {
         isObserved: true,
         lastSentiment: 'Indifferent',
+        observerName: 'Theresa Edwards',
         observer: {
```
Comment (jhweaver) [inline assets/js/mockData/interactionObservedData.js]: Actually, we got rid of the entire observer doc here. This whole subdoc should go away.

File: components/interactionEditor/contextBar/contextBar.js [hunk 5]
```diff
@@ -63,6 +64,19 @@ export default class ContextBar extends Component {
...
+        let interactionTime = objectPath.get(contextBarData, 'observation.observer.interactionStartDate');
```
Comment (jhweaver) [inline components/interactionEditor/contextBar/contextBar.js]: This is going away (and was redundant anyway). Use the startDate attribute on the document root. That one should always be there (regardless of interaction type)

File: test/components/interactionEditor/contextBar/contextBar.spec.js [hunk 6]
```diff
@@ -35,8 +35,8 @@ describe('Context Bar', () => {
     const observation = {
         isObserved: true,
         lastSentiment: 'Indifferent',
+        observerName: chance.word(),
         observer: {
```
Comment (jhweaver) [inline test/components/interactionEditor/contextBar/contextBar.spec.js]: Again, this whole subdocument is gone now.

File: components/interactionEditor/contextBar/contextBar.js (hunk 2 above)
Comment (troyblank) [inline components/interactionEditor/contextBar/contextBar.js]: Alright this has been updated, thanks.

File: assets/js/mockData/interactionObservedData.js (hunk 3 above)
Comment (troyblank) [inline assets/js/mockData/interactionObservedData.js]: Alright this has been updated, thanks.

**Pull request merged**